include/seqan3/std/memory

```diff
@@ -0,0 +1,68 @@
+// -*- C++ -*-
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides C++20 additions to the \<memory\> header.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#if __has_include(<version>)
+#include <version> // from C++20 all feature macros should be defined here
+#endif
+
+//!\brief A workaround for __cpp_lib_to_address for gcc version on macOS >=9.0 and < 9.4 (in C++20 mode).
+//!       Those versions implemented std::to_address, but did not define that feature detection macro.
+#ifndef SEQAN3_WORKAROUND_CPP_LIB_TO_ADDRESS
+#   if defined(__cpp_lib_to_address)
+#       define SEQAN3_WORKAROUND_CPP_LIB_TO_ADDRESS 1
+#   elif defined(__GNUC__) && defined(__APPLE__) && ((__GNUC__ == 9) && (__GNUC_MINOR__ < 4)) && __cplusplus > 201703L
```

Comment:
I wouldn't give an upper limit for the minor, because we know that the macos gcc 9 does not define that macro. 

Reason:
1. We know that macos gcc 9 is defining that function but forgot to add the version feature macro
2. If macos gcc 9 defines __cpp_lib_to_address in a later minor version it will use the first #2287 if case.